project_mis_report_budget/models/project_project.py

```diff
@@ -13,33 +13,36 @@ class ProjectProject(models.Model):
 
     def auto_create_mis_instance(self):
         super().auto_create_mis_instance()
-
         margin_kpi_ids = self.mis_report_template_id.kpi_ids.filtered(lambda x: x.kpi_type == 'margin')
         expression_id = margin_kpi_ids and margin_kpi_ids.expression_ids[0].id or False
 
         if expression_id:
-            budget_id = self.env['mis.budget'].create({
-                'name': '%s - %s' % (self.name, _('MARGINS')),
-                'project_id': self.id,
-                'report_id': self.mis_report_template_id.id,
-                'date_from': self.date_start.replace(day=1),
-                'date_to': self.date.replace(day=1) + relativedelta(months=2) - relativedelta(days=1),
-            })
+            budget_id = self.mis_budget_ids.filtered(lambda x: x.name == '%s - %s' % (self.name, _('MARGINS')))

```

Comment:
This is a very weak margins budget identification. E.g. if any user changes budget name before this operation, it won't work. Is it possible to add a technical field for budgets that indicate "Is a Margin Budget" and filter by it, and then create margin budget with such check before?

And, if name is finally used as "filtering" technique, please create a variable first with such value, as used twice (filtering and creating).